blocks/api/post.pegjs

```diff
@@ -1,33 +1,179 @@
+ * JavaScript for use the browser and one in PHP for
+ * WordPress itself. PEG parser generators are available
+ * in many languages, though different libraries may require
+ * some translation of this grammar into their syntax.
+ *
+ * For more information:
+ * @see https://pegjs.org
+ * @see https://en.wikipedia.org/wiki/Parsing_expression_grammar
+ *
+ */
+
 /** <?php
 // The `maybeJSON` function is not needed in PHP because its return semantics
 // are the same as `json_decode`
+
+// array arguments are backwards because of PHP
+if ( ! function_exists( 'peg_array_partition' ) ) {
+    function peg_array_partition( $array, callable $predicate ) {
+        $truthy = array();
+        $falsey = array();
+
+        foreach ( $array as $item ) {
+            call_user_func( $predicate, $item )
+                ? $truthy[] = $item
+                : $falsey[] = $item;
+        }
+
+        return array( $truthy, $falsey );
+    }
+}
+
+if ( ! function_exists( 'peg_join_blocks' ) ) {
+    function peg_join_blocks( $pre, $tokens, $post ) {
+        $blocks = array();
+        
+        if ( ! empty( $pre ) ) {
+            $blocks[] = array( 'blockName' => 'core/freeform', 'innerHTML' => $pre );
+        }
+        
+        foreach ( $tokens as $token ) {
+            list( $token, $html ) = $token;
+            
+            $blocks[] = $token;
+            
+            if ( ! empty( $html ) ) {
+                $blocks[] = array( 'blockName' => 'core/freeform', 'innerHTML' => $html );
+            }
+        }
+        
+        if ( ! empty( $post ) ) {
+            $blocks[] = array( 'blockName' => 'core/freeform', 'innerHTML' => $post );
+        }
+        
+        return $blocks;
+    }
+}
+
 ?> **/
 
+function freeform( s ) {
+  return s.length && {
+    blockName: 'core/freeform',
+    innerHTML: s
+  };
+}
+
+function joinBlocks( pre, tokens, post ) {
+    var blocks = [], i, l, html, item, token;
+    
+    if ( pre.length ) {
+        blocks.push( freeform( pre ) );
+    }
+    
+    for ( i = 0, l = tokens.length; i < l; i++ ) {
+        item = tokens[ i ];
+        token = item[ 0 ];
+        html = item[ 1 ];
+        
+        blocks.push( token );
+        if ( html.length ) {
+            blocks.push( freeform( html ) );
+        }
+    }
+    
+    if ( post.length ) {
+        blocks.push( freeform( post ) );
+    }
+    
+    return blocks;
+}
+
 function maybeJSON( s ) {
-	try {
-		return JSON.parse( s );
-	} catch (e) {
-		return null;
-	}
+    try {
+        return JSON.parse( s );
+    } catch (e) {
+        return null;
+    }
+}
+
+function partition( predicate, list ) {
+    var i, l, item;
+    var truthy = [];
+    var falsey = [];
+
+    // nod to performance over a simpler reduce
+    // and clone model we could have taken here
+    for ( i = 0, l = list.length; i < l; i++ ) {
+        item = list[ i ];
+
+        predicate( item )
+            ? truthy.push( item )
+            : falsey.push( item );
+    };
```

Comment:
Unnecessary semi-colon. Guessing it wouldn't be very easy for us to apply linting to this file, would it?

Reply:
is it? I thought I would be following style guidelines by including it.

Reply:
Yes, unnecessary: http://bit.ly/2y5BO9c